readthedocs/projects/tasks/builds.py

```diff
@@ -624,18 +625,21 @@ def collect_build_data(self):
         except Exception:
             log.exception("Error while collecting build data")
 
-    def upload_build_data(self):
+    def save_build_data(self):
         """
-        Upload data collected from the build after the build has ended.
+        Save the data collected from the build after it has ended.
 
         This must be called after the build has finished updating its state,
         otherwise some attributes like ``length`` won't be available.
         """
         try:
             if self.data.build_data:
-                api_v2.build(self.data.build_pk).telemetry.post(self.data.build_data)
+                save_build_data.delay(
+                    build_id=self.data.build_pk,
+                    data=self.data.build_data,
+                )
         except Exception:
-            log.exception("Error while uploading build data")
+            log.exception("Error while saving build data")
```

Comment:
Do we need a try/except here? Why the `.delay` would fail? It made more sense when using an API call, but I think we can remove this block now that it's using a Celery task.

Reply:
just being defensive here. Other reasons it could fail would be that self.data or its attributes aren't defined.